FirebaseAuth/Sources/Auth/FIRAuth.m

```diff
@@ -2171,6 +2171,12 @@ - (nullable NSString *)getUserID {
 
 - (BOOL)useUserAccessGroup:(NSString *_Nullable)accessGroup
                      error:(NSError *_Nullable *_Nullable)outError {
+  // self.storedUserManager is initialized asynchronously.
+  __block FIRAuthStoredUserManager *storedUserManager;
+  dispatch_sync(FIRAuthGlobalWorkQueue(), ^{
```

Comment:
This creates the potential for deadlock because `useUserAccessGroup:error:` is called from code already on the `FIRAuthGlobalWorkQueue()`

See https://github.com/firebase/firebase-ios-sdk/blob/f41bcaf177b7bca5f8987cdb382484e50c2072aa/FirebaseAuth/Sources/Auth/FIRAuth.m#L532

Reply:
Thanks @wilhuff Good catch! PTAL at the update and let me know if you have a suggestion for a better way.